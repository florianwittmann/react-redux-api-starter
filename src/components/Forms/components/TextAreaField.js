import React, { Component } from "react";
import { Field } from "redux-form";

class TextArea extends Component {
  field = props => {
    let { meta, title, placeholder } = props;

    return (
      <div className="field">
        <label className="label">{title}</label>
        <div className="control">
          <textarea
            className={
              meta.touched && meta.error ? "textarea is-danger" : "textarea"
            }
            {...props.input}
            placeholder={placeholder ? placeholder : ""}
          />

          {meta.touched && meta.error
            ? <span className="icon is-small is-right">
                <i className="fa fa-warning" />
              </span>
            : ""}
        </div>
        <p className="help is-danger">
          {meta.touched ? meta.error : ""}
        </p>
      </div>
    );
  };

  render() {
    const componentToRender = this.props.component;

    return (
      <Field {...this.props} {...componentToRender} component={this.field} />
    );
  }
}

export default TextArea;
