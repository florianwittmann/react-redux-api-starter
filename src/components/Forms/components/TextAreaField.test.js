import React from "react";
import { shallow, render } from "enzyme";
import TextAreaField from "./TextAreaField";

test("should wrap redux-form field correctly", () => {
  const wrapper = shallow(<TextAreaField name="test" title="Area Title" />);
  const field = wrapper.prop("component");
  expect(field).toBeInstanceOf(Function);
  expect(wrapper).toMatchSnapshot();
});
