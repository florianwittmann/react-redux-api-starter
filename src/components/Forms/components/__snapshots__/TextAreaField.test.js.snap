// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should wrap redux-form field correctly 1`] = `
<Field
  component={[Function]}
  name="test"
  title="Area Title"
/>
`;
